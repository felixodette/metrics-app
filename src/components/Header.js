import React from 'react';
import PropTypes from 'prop-types';
import styles from './Header.module.css';

const Header = (props) => {
  const { heading } = props;
  return (
    <header>
      <i className={styles.previous} />
      <h1>{ heading }</h1>
      <div className={styles.icons}>
        <i className={styles.mic} />
        <i className={styles.settings} />
      </div>
    </header>
  );
};

export default Header;

Header.propTypes = {
  heading: PropTypes.string.isRequired,
};
