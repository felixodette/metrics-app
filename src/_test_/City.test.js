import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Header from '../components/Header';

describe('Create header', () => {
  it('Render without errors', () => {
    const tree = renderer
      .create(
        <Router>
          <Header heading="test" previous="" />
        </Router>,
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the elements', () => {
    render(
      <Router>
        <Header heading="test" previous="" />
      </Router>,
    );
    const element = screen.getByText('testu');
    expect(element).toBeInTheDocument();
  });
});