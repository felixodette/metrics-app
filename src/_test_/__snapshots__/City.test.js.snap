// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create header Render without errors 1`] = `
<header>
  <span />
  <h3>
    test
  </h3>
  <div
    className="icons"
  >
    <i
      className="mic"
    />
    <i
      className="settings"
    />
  </div>
</header>
`;
